security/cmd/istio_ca/main.go

```diff
@@ -201,30 +211,34 @@ func createClientset() *kubernetes.Clientset {
 }
 
 func createCA(core corev1.SecretsGetter) ca.CertificateAuthority {
+	var caOpts *ca.IstioCAOptions
+	var err error
+
 	if opts.selfSignedCA {
 		log.Info("Use self-signed certificate as the CA certificate")
 
 		// TODO(wattli): Refactor this and combine it with NewIstioCA().
-		istioCA, err := ca.NewSelfSignedIstioCA(opts.caCertTTL, opts.workloadCertTTL, opts.maxWorkloadCertTTL, opts.selfSignedCAOrg,
-			opts.istioCaStorageNamespace, core)
+		caOpts, err = ca.NewSelfSignedIstioCAOptions(opts.caCertTTL, opts.workloadCertTTL,
+			opts.maxWorkloadCertTTL, opts.selfSignedCAOrg, opts.istioCaStorageNamespace, core)
 		if err != nil {
 			fatalf("Failed to create a self-signed Istio CA (error: %v)", err)
 		}
-		return istioCA
+	} else {
+		var certChainBytes []byte
```

Comment:
To make it consistent with the code above, I think you can refactor the following code to ca.NewIstioCAOptions(). You can add a TODO here.

Reply:
Created https://github.com/istio/istio/issues/2977 for this.